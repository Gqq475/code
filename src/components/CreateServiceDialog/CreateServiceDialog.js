import React from 'react'
import classes from './CreateServiceDialog.scss'
import DialogFilter from 'components/DialogFilter'
import { connect } from 'react-redux'
const qca = __QCA__

type Props = {
 data: Object,
 isOpen: boolean,
 toggleDialog: Function,
 getModel: Function,
 getCreateService: Function,
 dataList: Object,
 token: String
};
export class CreateServiceDialog extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.getKernelIds = this.getKernelIds.bind(this)
    this.kernelIdFilter = this.kernelIdFilter.bind(this)
    this.loadingKernelIds = false
    this.state = {
      name: '',
      modelId: undefined,
      isChanged: false,
      kernelIds: [],
      kernelId: undefined
    }
  }

  handleSubmit () {
    var tmp = {
      name: this.state.name,
      deviceId: '1',
      modelId: this.state.modelId,
      dataSource: '/image/T00B351BP/',
      description: 'Test',
      kernelId: this.state.kernelId
    }
    this.props.getCreateService(tmp)
    this.props.toggleDialog()
    this.setState({
      isChanged: false
    })
  }

  kernelIdFilter () {
    let kernelIds = []
    let propKernelIds = []
    let resultKernelIds = this.state.kernelIds
    // data filter
    this.props.dataList &&
    this.props.dataList.length &&
    this.props.dataList.forEach((item) => {
      propKernelIds.push(item.kernelId)
    })
    if (propKernelIds.length) {
      resultKernelIds.forEach((kernelId) => {
        !propKernelIds.includes(+kernelId) && kernelIds.push(+kernelId)
      })
    } else {
      kernelIds = resultKernelIds
    }

    return kernelIds
  }

  getKernelIds () {
    this.loadingKernelIds = true
    fetch(`${qca}/ojs/service/kernels`, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': this.props.token
      }
    })
    .then(response => response.json())
    .then((data) => {
      this.loadingKernelIds = false
      if (data.code === 200000) {
        this.setState({
          kernelIds: data.result
        })
      } else {
        console.log('kernelId 获取失败')
      }
    })
  }

  componentWillMount () {
    this.props.getModel()
    this.getKernelIds()
  }

  render () {
    return (
      <div className={classes['CreateServiceDialog-container']}>
        <DialogFilter
          DialogTitle='Create Service'
          DialogID=''
          DialogName=''
          DialogBtnContent='Create'
          isOpen={this.props.isOpen}
          toggleDialog={this.props.toggleDialog}
          handleSubmit={this.handleSubmit}
          getModel={this.props.getModel}
          data={this.props.data}
          getCreateService={this.props.getCreateService}
          isChanged={this.state.isChanged}
          that={this}
          modelId={this.state.modelId}
          kernelIds={this.kernelIdFilter()}
          kernelId={this.state.kernelId}
          loadingKernelIds={this.loadingKernelIds} />
      </div>
    )
  }
}
export default connect(
  (state) => ({
    token: state.auth.token
  }),
  (dispatch, state) => {
    return {
      dispatch
    }
  }
)(CreateServiceDialog)
