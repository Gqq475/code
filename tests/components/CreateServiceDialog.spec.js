import React from 'react'
import { CreateServiceDialog } from 'components/CreateServiceDialog/CreateServiceDialog'

describe('(Component) CreateServiceDialog', () => {
  it('render normally', () => {
    const getModel = jest.fn()
    const wrapper = shallow(
      <CreateServiceDialog getModel={getModel} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
