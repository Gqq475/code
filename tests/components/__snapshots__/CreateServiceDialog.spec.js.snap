// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) CreateServiceDialog render normally 1`] = `
<div
  className="CreateServiceDialog-container"
>
  <DialogFilter
    DialogBtnContent="Create"
    DialogID=""
    DialogName=""
    DialogTitle="Create Service"
    getModel={[Function]}
    handleSubmit={[Function]}
    isChanged={false}
    kernelIds={Array []}
    loadingKernelIds={true}
    that={
      CreateServiceDialog {
        "_reactInternalInstance": ShallowComponentWrapper {
          "_calledComponentWillUnmount": false,
          "_compositeType": 0,
          "_context": Object {},
          "_currentElement": <CreateServiceDialog
            getModel={[Function]}
      />,
          "_debugID": 1,
          "_hostContainerInfo": null,
          "_hostParent": null,
          "_instance": [Circular],
          "_mountOrder": 1,
          "_pendingCallbacks": null,
          "_pendingElement": null,
          "_pendingForceUpdate": false,
          "_pendingReplaceState": false,
          "_pendingStateQueue": null,
          "_renderedComponent": NoopInternalComponent {
            "_currentElement": <div
              className="CreateServiceDialog-container"
      >
              <DialogFilter
                      DialogBtnContent="Create"
                      DialogID=""
                      DialogName=""
                      DialogTitle="Create Service"
                      data={undefined}
                      getCreateService={undefined}
                      getModel={[Function]}
                      handleSubmit={[Function]}
                      isChanged={false}
                      isOpen={undefined}
                      kernelId={undefined}
                      kernelIds={Array []}
                      loadingKernelIds={true}
                      modelId={undefined}
                      that={[Circular]}
                      toggleDialog={undefined}
              />
      </div>,
            "_debugID": 2,
            "_renderedOutput": <div
              className="CreateServiceDialog-container"
      >
              <DialogFilter
                      DialogBtnContent="Create"
                      DialogID=""
                      DialogName=""
                      DialogTitle="Create Service"
                      data={undefined}
                      getCreateService={undefined}
                      getModel={[Function]}
                      handleSubmit={[Function]}
                      isChanged={false}
                      isOpen={undefined}
                      kernelId={undefined}
                      kernelIds={Array []}
                      loadingKernelIds={true}
                      modelId={undefined}
                      that={[Circular]}
                      toggleDialog={undefined}
              />
      </div>,
          },
          "_renderedNodeType": 0,
          "_rootNodeID": 0,
          "_topLevelWrapper": null,
          "_updateBatchNumber": null,
          "_warnedAboutRefsInRender": false,
        },
        "context": Object {},
        "getKernelIds": [Function],
        "handleSubmit": [Function],
        "kernelIdFilter": [Function],
        "loadingKernelIds": true,
        "props": Object {
          "getModel": [Function],
        },
        "refs": Object {},
        "state": Object {
          "isChanged": false,
          "kernelId": undefined,
          "kernelIds": Array [],
          "modelId": undefined,
          "name": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      }
    }
  />
</div>
`;
